src/hotspot/cpu/x86/stubGenerator_x86_32.cpp

```diff
@@ -3669,17 +3669,32 @@ class StubGenerator: public StubCodeGenerator {
     __ pusha();
 
     // xmm0 and xmm1 may be used for passing float/double arguments
-    const int xmm_size = wordSize * 4;
-    const int xmm_spill_size = xmm_size * 2;
-    __ subptr(rsp, xmm_spill_size);
-    __ movdqu(Address(rsp, xmm_size * 1), xmm1);
-    __ movdqu(Address(rsp, xmm_size * 0), xmm0);
+
+    if (UseSSE >= 2) {
+      const int xmm_size = wordSize * 4;
+      __ subptr(rsp, xmm_size * 2);
+      __ movdbl(Address(rsp, xmm_size * 1), xmm1);
+      __ movdbl(Address(rsp, xmm_size * 0), xmm0);
+    } else if (UseSSE >= 1) {
+      const int xmm_size = wordSize * 2;
+      __ subptr(rsp, xmm_size * 2);
+      __ movflt(Address(rsp, xmm_size * 1), xmm1);
+      __ movflt(Address(rsp, xmm_size * 0), xmm0);
+    }
```

Comment:
I am not sure that we should sneak in this shared-code change with what is otherwise a Shenandoah-change.

Reply:
This is nmethod barrier handling part, and on x86_32, Shenandoah is the only user of this code. I can split it out, but it would still be Shenandoah-specific.

Reply:
Ah ok then.

Reply:
(...and I don't like to split it, because both hunks are needed to pass testing)